src/Logging/Logging/src/FilterLoggingBuilderExtensions.cs

```diff
@@ -11,58 +11,181 @@ namespace Microsoft.Extensions.Logging
     /// </summary>
     public static class FilterLoggingBuilderExtensions
     {
+        /// <summary>
+        /// Adds a log filter to the factory.
+        /// </summary>
+        /// <param name="builder">The <see cref="ILoggingBuilder"/> to add the filter to.</param>
+        /// <param name="filter">The filter to be added.</param>
```

Comment:
These freakin' callbacks of type `Func<string, string, LogLevel, bool>` are not clear at all. What are the 3 params there? Category, *something*, and LogLevel?

Reply:
Category and Provider (Console/Debug/AzureAppService/EventSource/Etc.) are the two, I don't know which order though. At some point with these, `Func<,,,,>` loses it's charm...